### ISSUE TITLE
"Referencing the Window in Angular expressions is disallowed" error thrown without any reference to window

### ISSUE THREAD

**Issue by shank8** [angular/angular.js#10685]
I have a form in my application that has been working just fine with AngularJS v1.2.17 for weeks now.  I recently tried to upgrade to v1.2.28, but noticed that this form now throws JavaScript errors even though the code has not changed.

I have a question on StackOverflow regarding [this issue](http://stackoverflow.com/questions/27848457/referencing-the-window-in-angular-expressions-is-disallowed-occurs-when-not-u)

I realized that the error gets thrown in v1.2.19 and above. I have absolutely NO references to the window object in either my template expressions or my controllers. Therefore, I have no idea what caused this issue to bubble up in these newer builds.

**Comment by lgalfaso** [comment]
Hi. With only the information above, it is not enough to know what is going on. Would it be possible for you to create a plunker that shows this issue?

**Comment by shank8** [comment]
I have no idea what is actually causing it so I can't specify what code needs to go into a plunker.  Seeing how other people haven't had this problem I am guessing it is an environment problem, but that is just a guess. If I try to use the same form in plunker everything works fine. This is why it is so weird... because Angular is appearing to throw exceptions on lines of code that have nothing to do with the given error.

**Comment by lgalfaso** [comment]
Hi, is your application available somewhere? Without a minimal way to be able to see the error, there is close to no chances for this to get solved

**Comment by shank8** [comment]
Unfortunately no, it is part of an unreleased product that I am legal not allowed to share.  I wish there were a way for me to show you the error. The thrown exception makes absolutely no sense with what any of my code is doing and the fact that the error is introduced in 1.2.19+ makes me very curious

**Comment by shank8** [comment]
Could I send a private message of some sort to you with screen shots/code fragments that may help?

**Comment by lgalfaso** [comment]
I am sorry, but without a way to reproduce the issue, not having an expression/scope that triggers this nor being able to try this out, then there is nothing else that can be done. I am going to close this issue, but if you find a way to reproduce this with an example that you can out at plunker, feel free to reopen the issue

**Comment by shank8** [comment]
In my StackOverflow I state exactly what expression the error is supposedly occurring on, however you can see that the expression has nothing to do with Window in any way.  I can't pinpoint what area of the code to look at when the error gives me no indication of what is happening.

**Comment by pkozlowski-opensource** [comment]
@shank8 it is very hard (not to say impossible) to identify / fix the issue you are facing. Try to narrow down the code that is failing and reproduce it in an isolated manner in a jsfiddle, plunker or a similar tool. If you do this I'm sure that you are going to get an answer in no time!

**Comment by lgalfaso** [comment]
the way angular checks if some object is `window` is with the check
```js
function isWindow(obj) {
  return obj && obj.document && obj.location && obj.alert && obj.setInterval;
}
```

the only possibility that I can think of is that some object you are referencing has all these properties

**Comment by shank8** [comment]
Ok thanks for the info. I will continue to try and reproduce/debug the issue

**Comment by shank8** [comment]
Ah... I think I found the problem.  After traversing the stack trace I realized that the object that threw the error was my `$scope.model` object which is what I use in all my controllers as the root property in which all other properties extend from.

I realized that in my controller I typically set up my `$scope.model` by initializing it to some OOP Class like

```
function HomeViewModel() {
 this.title = "Hello";
 this.content = "World";
}
...
$scope.model = new HomeViewModel();
```

This should create a new instance of this HomeViewModel class, however I realized that in this one controller I accidentally had

```
$scope.model = HomeViewModel();
```

So I forgot the `new` operator which caused the `this` property within my class to reference `window`.. therefore, my `$scope.model` was equal to `window` extended with my actual model.  So the `$scope.model` then caused the exception to be thrown since it contains the defined properties that @lgalfaso stated above.

So now my question is why was this allowed before? My guess is there was never a restriction on using the `window` object prior to version 1.2.19? That would make the most sense.

Another question I have is why the error was thrown on this particular expression. There are many expression preceding this one that use `$scope.model` so I am not sure why it happened to be this one
